### PR TITLE
Improve logic of table output file creation

### LINKED ISSUE #24: Improve table add_image function
Currently, a call to `Table.add_image` immediately triggers creation of output files. As a consequence, you need to pass the submission output folder to the `Table.add_image` function.

Probably better:
* Change `Table.add_image` to only store the path of the image in the Table
* Add `Table.create_image_files`  function that writes thumbnails, etc for all image paths stored in the Table.
* Rename `Table.write_yaml` to `Table.create_yaml_files`
* Create new `Table.create_files` function that calls `Table.create_image_files` and `Table.create_yaml_files`
* Finally, replace call to `Table.write_yaml` in `Submission.create_files` with a call to `Table.create_files`. The output directory can be passed all the way down.

### PR DESCRIPTION
**Pull request by AndreasAlbert**
Closes #24

### CODE REVIEW

**Comment by AndreasAlbert** [comment]
Don't worry about reviewing, this is still buggy. #30  can be reviewed/merged, IMO, though

**Comment by AndreasAlbert** [comment]
Ok, ready to be merged at your convenience. Can either merge #30 first and then #31 or only merge #31 and close #30. Result will be identical.

File: hepdata_lib/__init__.py [hunk 1]
```diff
@@ -657,6 +745,8 @@ def get_hist_2d_points(hist):
         for y_bin in range(1, hist.GetNbinsY() + 1):
             y_val = hist.GetYaxis().GetBinCenter(y_bin)
             z_val = hist.GetBinContent(x_bin, y_bin)
+            dz_val = hist.GetBinError(x_bin, y_bin)
```
Comment (clelange) [inline hepdata_lib/__init__.py]: Same comment as for #30 - what about asymmetric errors?

File: tests/test_submission.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from hepdata_lib import Submission, Table, Variable, Uncertainty
+
+class TestSubmission(TestCase):
+    """Test the Submission class."""
+    def test_add_table(self):
+        """Test the add_table function."""
+
+        # Verify that the type check works
+        test_submission = Submission()
+        test_table = Table("Some Tale")
```
Comment (clelange) [inline tests/test_submission.py]: typo Tale -> Table

File: hepdata_lib/__init__.py [hunk 3]
```diff
@@ -666,10 +756,42 @@ def get_hist_2d_points(hist):
             points["y_edges"].append((y_val - width_y / 2, y_val + width_y / 2))
 
             points["z"].append(z_val)
+            points["dz"].append(dz_val)
 
     return points
 
 
+
+def get_hist_1d_points(hist):
```
Comment (clelange) [inline hepdata_lib/__init__.py]: I think this is part of #30 - won't review here.

File: tests/test_submission.py (hunk 2 above)
Comment (AndreasAlbert) [reply tests/test_submission.py]: Fixed

File: hepdata_lib/__init__.py [hunk 4]
```diff
@@ -205,35 +209,88 @@ def __init__(self, name):
...
+        Add an image file to the table.
+
+        This function only stores the path to the image.
+        Any additional processing will be done later
+        (see write_images function).
+
+        :param file_path: Path to the image file.
+        :type file_path: string
+
+        :param outdir: Deprecated.
```
[the remark this answers is not in the document]
Comment (AndreasAlbert) [reply hepdata_lib/__init__.py]: I know of at least two people actively using it ATM. Would prefer to keep it for a while and then remove later.

**Pull request merged**